ext/pg_query/pg_query_parse.c

```diff
@@ -15,7 +15,11 @@ VALUE new_parse_error(ErrorData* error)
 	cPgQuery = rb_const_get(rb_cObject, rb_intern("PgQuery"));
 	cParseError = rb_const_get_at(cPgQuery, rb_intern("ParseError"));
 
-	args[0] = rb_str_new2(error->message);
+	char *message = NULL;
+	message = malloc(snprintf(NULL, 0, "%s (%s:%d)", error->message, error->filename, error->lineno) + 1);
+	sprintf(message, "%s (%s:%d)", error->message, error->filename, error->lineno);
```

Comment:
Nice, good idea!

I'm actually working on extracting the C component of this library (so its re-usable), and have taken a struct-based approach there: https://github.com/lfittl/libpg_query/blob/master/pg_query.h#L7 & https://github.com/lfittl/libpg_query/blob/master/pg_query_parse.c#L75

To support that effort, I think it might be nicer to pass this back as `args[2]` and `args[3]` so the Ruby part can decide what to do with it.